.circleci/config.yml

```
@@ -0,0 +1,33 @@
+version: 2.1
+
+orbs:
+  aws-ecr: circleci/aws-ecr@7.3.0
+
+jobs:
+  build-and-push-image:
+    parameters:
+      postgres-version:
+        type: string
+    docker:
+      - image: cimg/base:stable
+    steps:
+      - checkout
+      - setup_remote_docker:
+          version: 19.03.13
+      - aws-ecr/build-and-push-image:
+          account-url: AWS_ECR_ACCOUNT_URL
+          aws-access-key-id: AWS_ACCESS_KEY_ID
+          aws-secret-access-key: AWS_SECRET_ACCESS_KEY
+          region: AWS_ECR_DEFAULT_REGION
+          repo: ${CIRCLE_PROJECT_REPONAME}
+          extra-build-args: "--rm --build-arg POSTGRES_VERSION=<<parameters.postgres-version>>"
+          tag: <<parameters.postgres-version>>
```

ain't this always override an existing tag, presumably stable, potentially opening the door for silent failures?

can we add the git hash (maybe in short version or not) to make it unique?

---

that comes from the matrix parameters.  if it was unset, circleci would not fail silently.

this is an input, not an output, so adding a git hash would be pretty painful. :)

---

(sorry, just realized this was on the tag: line.  github's line wrap make it look like it was on the extra-build-args line)

yeah, we need a bit better handling here.   only `v2` (aka our main branch) should push a a raw tag.   we don't want random branches and PRs to push that.

we haven't really defined a tagging strategy for this yet.

---

we could use the same idea that https://github.com/AlayaCare/db-migrations.base/blob/v2/.circleci/config.yml uses - 

`tag: "${GIT_BRANCH_SLUG}"-<<...version>>`  

that would prefix any push with the branch - and we'd normally only consume `v2-version` tags but would have `branch-version` tags available for testing.

I think we _would_ want to overwrite an existing tag, since we're not changing upstream images, only the content we're pushing into it?   

Else we need a much more defined tagging strategy.

---

What about adding the branch name as prefix? Example:

v2-9.6